### DISCUSSION TITLE
What's the right way to get all the coordinates to set the maxBounds?

### DISCUSSION THREAD

**Discussion by eduardomarr** [uber/react-map-gl#1777] [General]
What's the right way to get all the coordinates to set the maxBounds?

I do this and it doesn't work:

if my restrict area is some like that..

![Captura de Tela 2022-02-24 às 13 11 51](https://user-images.githubusercontent.com/24718475/155563969-e9efba7f-01ad-4427-8da0-5e6d17204cd0.png)

What I usually do is, take the map and go a little further down and get the longitude and latitude data to use on "Southwest coordinates" array.
![Captura de Tela 2022-02-24 às 13 12 22](https://user-images.githubusercontent.com/24718475/155566742-c85795ce-d139-4f69-b652-5b50bca45dcd.png)

I do the same on North:

![Captura de Tela 2022-02-24 às 13 12 31](https://user-images.githubusercontent.com/24718475/155567053-6133bde1-712d-4abf-bb58-98a94dba258c.png)

my property would look like this:
```
 maxBounds={[
        [-45.0873499, -23.4478651], // Southwest coordinates
        [-45.0869838, -23.4271727] // Northeast coordinates
      ]}
```

if i use all of this coordinates in my component, this will bw the result:

```
 return (
    <Map
      {...viewport}
      initialViewState={{
        longitude: -45.0870243,
        latitude: -23.4370723,
      }}
      maxBounds={[
        [-45.0873499, -23.4478651], // Southwest coordinates
        [-45.0869838, -23.4271727] // Northeast coordinates
      ]}
      style={{ width: '60vw', height: '60vh' }}
      mapboxAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
      mapStyle={process.env.REACT_APP_MAPBOX_STYLE}
      onMove={onViewportChange}
    />
  );
```

![Captura de Tela 2022-02-24 às 13 39 07](https://user-images.githubusercontent.com/24718475/155568008-e1507feb-2c86-4542-a777-d9ca4b58840d.png)

Someone could help me?

**Comment by saintsalad** [comment]
@eduardomarr Do you solved this?

**Comment by joel-daros** [comment]
The coordinates inside bound needs to follow this format:

```
  [
    [boundsObject.ne.lon, boundsObject.ne.lat],
    [boundsObject.sw.lon, boundsObject.sw.lat],
  ];
```

**Reply by saintsalad** [reply]
I tried this 
`  const maxBounds = [
    [121.7829649716893, 17.04453545413898],
    [122.2914574628461, 16.636550701667574]
  ]`
and this
`  const maxBounds = [
    { lon: 121.7829649716893, lat: 17.04453545413898 },
    { lon: 122.2914574628461, lat: 16.636550701667574 }
  ]`

but I got this error, I am very confuse with this 🥲

`Uncaught Error: Invalid LngLat object: (NaN, NaN)`

**Reply by saintsalad** [reply]
Can you please provide a good example?

**Reply by joel-daros** [reply]
```
// Set bounds to San Francisco, California.
const bounds = [
[-122.66336, 37.492987], // Southwest coordinates
[-122.250481, 37.871651] // Northeast coordinates
];
```

Check a working example here: https://docs.mapbox.com/mapbox-gl-js/example/restrict-bounds/
